### PR TITLE
various text fixes

### PR DESCRIPTION
**Pull request by tshepang**

### CODE REVIEW

File: ferrocene/doc/user-manual/src/cli.rst [hunk 1]
```diff
@@ -166,8 +166,9 @@ Command-Line Interface
 
       .. caution::
 
-         Only the arguments listed in section Compilation targets are within
-         the scope of the Ferrocene |iso_ref| qualification.
+         This code generation option is outside the scope of the Ferrocene
```
Comment (pietroalbini) [inline ferrocene/doc/user-manual/src/cli.rst:169]: This is not correct, and the old text should stay: this means, you *can* use `-C link-arg`, but only with the values we describe in each target's documentation.

Besides, we should remove |iso_ref| from all of these cautions.

Comment (pietroalbini) [reply ferrocene/doc/user-manual/src/cli.rst]: This needs to be applied to the other similar changes too, it's not resolved.

File: ferrocene/doc/user-manual/src/cli.rst [hunk 2]
```diff
@@ -241,11 +241,6 @@ Command-Line Interface
 
    .. cli:option:: -C linker=<path>
 
-      .. caution::
-
-         Only the paths listed in section Compilation targets are within the
-         scope of the Ferrocene |iso_ref| qualification.
```
Comment (pietroalbini) [inline ferrocene/doc/user-manual/src/cli.rst:247]: We should say this that it is possible to use it only for targets that require a linker driver, and in that case the linker driver must respect the option.

Comment (tshepang) [reply ferrocene/doc/user-manual/src/cli.rst]: an attempt https://github.com/ferrocene/ferrocene/pull/324/commits/74af1d982dd352c1ebc3f35521dd94df8f99f791

File: ferrocene/doc/user-manual/src/cli.rst [hunk 3]
```diff
@@ -241,6 +241,11 @@ Command-Line Interface
 
    .. cli:option:: -C linker=<path>
 
+      .. caution::
+
+         This can only be used for targets that require a linker driver,
+         in which case the linker driver must respect the option.
```
Comment (pietroalbini) [inline ferrocene/doc/user-manual/src/cli.rst:247]: Can this link to the safety manual?

**Review by pietroalbini — APPROVED** [review]
bors merge

**Pull request merged**